core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -631,4 +651,112 @@ public ShareFetchPartitionData(FetchParams fetchParams, String groupId, String m
             this.partitionMaxBytes = partitionMaxBytes;
         }
     }
+
+    public static class ShareGroupMetrics {
```

Comment:
Do we want a public class as inner class or it can live separately in `server/share` package?

Reply:
we can make it non-public